Bulkloader: support for multiple values

peterstace opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/1498)
<!-- Reviewable:end -->

pawanrawal commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[cmd/bulkloader/systest/suite03/query.json, line 6 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498#-KuSqpDlMl-Sz8_Wcz2x:-KuSqpDlMl-Sz8_Wcz2y:bohrdaw) ([raw file](https://github.com/dgraph-io/dgraph/blob/6d1f12bbcb93d5429736aa6934102f93ea273079/cmd/bulkloader/systest/suite03/query.json#L6)):*
> ```JSON
> 	}
> 	occupation_count(func: has(occupation)) {
> 		name,
> ```

You don't need the comma here. Its needed if you want to have `count(occupations)` on the same line.

---

*[cmd/bulkloader/systest/suite03/query.json, line 12 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498#-KuSqfFv--AUmCIMgTCx:-KuSqfFv--AUmCIMgTCy:b-bgm48d) ([raw file](https://github.com/dgraph-io/dgraph/blob/6d1f12bbcb93d5429736aa6934102f93ea273079/cmd/bulkloader/systest/suite03/query.json#L12)):*
> ```JSON
> 		name
> 	}
> 	alex(func: anyofterms(name, "Alex"@.)) {
> ```

It should be `name@.`. Strange that this works. Looks like a bug. The parser should throw an error for this query.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498#-:-KuSrNV-gQ2iLqf2wBxw:bnfp4nl)*
<!-- Sent from Reviewable.io -->

peterstace commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[cmd/bulkloader/systest/suite03/query.json, line 6 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498#-KuSqpDlMl-Sz8_Wcz2x:-KuSv3uJoXbAomo6MyDx:bss4a2b) ([raw file](https://github.com/dgraph-io/dgraph/blob/6d1f12bbcb93d5429736aa6934102f93ea273079/cmd/bulkloader/systest/suite03/query.json#L6)):*
<details><summary><i>Previously, pawanrawal (Pawan Rawal) wrote…</i></summary><blockquote>

You don't need the comma here. Its needed if you want to have `count(occupations)` on the same line.
</blockquote></details>

Removed.

---

*[cmd/bulkloader/systest/suite03/query.json, line 12 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498#-KuSqfFv--AUmCIMgTCx:-KuSv5JLCWwywdPXkb2D:b7q9p25) ([raw file](https://github.com/dgraph-io/dgraph/blob/6d1f12bbcb93d5429736aa6934102f93ea273079/cmd/bulkloader/systest/suite03/query.json#L12)):*
<details><summary><i>Previously, pawanrawal (Pawan Rawal) wrote…</i></summary><blockquote>

It should be `name@.`. Strange that this works. Looks like a bug. The parser should throw an error for this query.
</blockquote></details>

Yep, it seems to work with both `anyofterms(name@., "Alex")` or `anyofterms(name, "Alex"@.)`. I've changed to use the former.

I opened an issue, https://github.com/dgraph-io/dgraph/issues/1499

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/1498)*
<!-- Sent from Reviewable.io -->
